src/v/redpanda/admin_server.cc

```diff
@@ -213,6 +212,43 @@ void admin_server::log_level_timer_handler() {
     rearm_log_level_timer();
 }
 
+ss::future<ss::httpd::redirect_exception>
+admin_server::redirect_to_leader(ss::httpd::request& req) const {
+    auto leader_id_opt = _metadata_cache.local().get_leader_id(
+      model::controller_ntp);
+
+    if (!leader_id_opt.has_value()) {
+        throw ss::httpd::base_exception(
+          fmt::format("Controller leader not available"),
+          ss::httpd::reply::status_type::service_unavailable);
+    }
+    auto leader_opt = _metadata_cache.local().get_broker(leader_id_opt.value());
+
+    if (!leader_opt.has_value()) {
+        throw ss::httpd::base_exception(
+          fmt::format("Controller leader metadata not available"),
+          ss::httpd::reply::status_type::service_unavailable);
+    }
+
+    // FIXME: We assume that our peers are listening on the same admin port as
+    // we are.
+    auto port = config::node_config().admin()[0].address.port();
```

Comment:
The port part of it holds up (k8s/docker-compose give each container its own IP, so all the containers typically use the same default port).

Advertised endpoint is a good call -- currently using the internal RPC address but it would make more sense to use the advertised one.

Reply:
Ah, never mind - we are already using advertised addresses because they're what go into the `model::broker` when a node registers itself.

Reply:
sounds good. maybe it doesn't work for every scenario, but we can add in more context to fix those when they arise?